core/src/main/java/com/linecorp/armeria/client/DnsResolverGroupBuilder.java

```diff
@@ -342,7 +354,9 @@ RefreshingAddressResolverGroup build(EventLoopGroup eventLoopGroup) {
                 builder.decodeIdn(decodeIdn);
             }
         };
+        cacheSpec = requireNonNull(cacheSpec != null ? cacheSpec : Flags.dnsCacheSpec(), "cacheSpec");
```

Comment:
nit: How about declaring a final local variable rather than modifying a member field? A user did not really set the `cacheSpec`.

```suggestion
        final String cacheSpec = firstNonNull(this.cacheSpec, Flags.dnsCacheSpec());
```